server/policies/document.js

```diff
@@ -122,6 +122,7 @@ allow(User, "delete", Document, (user, document) => {
 
 allow(User, "restore", Document, (user, document) => {
   if (!document.deletedAt) return false;
+  if (document.collection.deletedAt) return false;
```

Comment:
if the document does not have a collection, it's perma-deleted

Reply:
I'll probably merge that PR tomorrow so might as well just skip this check, what made you put it in rn?

Reply:
in this current state it would error out if you try to restore a document from a deleted collection